docs/NetEbpfExtMultiAttach.md

```diff
@@ -0,0 +1,126 @@
+# Introduction
+
+Multi-attach support allows multiple eBPF programs to be attached to the same attach point (aka "hook") with the
+same attach parameters. For example, for `BPF_CGROUP_INET4_CONNECT` attach type, the attach parameter is the
+compartment ID. With this feature, multiple eBPF programs can be attached to the same compartment ID.
+
+This document describes the design changes required in `netebpfext` to support multiple attach of eBPF programs in
+`netebpfext`. This is independent of the platform changes needed in BPF runtime to add support for new multi-attach
+APIs and flags. These changes in netebpfext also help enable the "default" behavior of multi-attach feature where the
+multiple legacy apps using older libbpf APIs to attach programs can still attach to the same hook, with the programs
+being "appended" in the invoke list in the order they are attached.
+
+# Current Design
+
+`net_ebpf_extension_hook_provider_t` is the provider context for the hook NPI provider. It maintains a list of client
+contexts. A client context represents the eBPF program that is attached to this specific "attach point". It also holds
+the attach params with which the program was attached. For each client context, a filter context is created. The filter
```

Comment:
```suggestion
the attach parameters with which the program was attached. For each client context, a filter context is created. The filter
```
Many times this file has attach params and other times attach parameters.  Pick one and be consistency throughout.  I'd probably prefer parameters under the principle of avoiding unnecessary abbreviations.

Also sometimes "attach params" has quotes like here, and sometimes it doesn't, like line 56. Be consistent. I'd suggest maybe use quotes the very first occurrence in the file and not thereafter.